### PR TITLE
Lab Adapter

### PR DESCRIPTION
**Pull request by mgorski98**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/JobsToDrawerAdapter.java [hunk 1]
```diff
@@ -27,11 +27,13 @@ public void operateTo(int x, int y) {
 		line.setStartCoordinates(this.startX, this.startY);
 		line.setEndCoordinates(x, y);
 
-		drawLine(line);
+		setPosition(x, y);
+		DrawerFeature.getDrawerController().drawLine(line);
```
Comment (rsusik) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/JobsToDrawerAdapter.java:31]: Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/JobsToDrawerAdapter.java [hunk 2]
```diff
@@ -4,14 +4,14 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class SquareDrawerAdapter implements Job2dDriver {
```
Comment (rsusik) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/JobsToDrawerAdapter.java:12]: The name does not indicate the purpose of the adapter or what was actually adapted.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        this.x = i;
+        this.y = i1;
+    }
+
+    @Override
+    public void operateTo(int i, int i1) {
+        line.setStartCoordinates(this.x, this.y);
+        line.setEndCoordinates(i, i1);
+        setPosition(i, i1);
+        DrawerFeature.getDrawerController().drawLine(line);
```
Comment (rsusik) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:27]: Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java [hunk 4]
```diff
@@ -16,6 +16,13 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
...
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+		switch (e.getActionCommand()) {
+			case "Figure Joe 1":
+				FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+				break;
+			case "Figure Joe 2":
+				FiguresJoe.figureScript2(driverManager.getCurrentDriver());
+				break;
```
Comment (rsusik) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:25]: Menu action should not depend on its caption. Enum or at least constants should be considered.

**Review by rsusik — COMMENTED** [review]
Specific remarks are inlined in the code.
The UML diagram class names do not match the names in source code.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java (hunk 3 above)
Comment (mgorski98) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:27]: Should be fixed now

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/JobsToDrawerAdapter.java (hunk 1 above)
Comment (mgorski98) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/JobsToDrawerAdapter.java:31]: Fixed

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/JobsToDrawerAdapter.java (hunk 2 above)
Comment (mgorski98) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/JobsToDrawerAdapter.java:12]: Also fixed (now named Jobs2dToDrawerAdapter)

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java (hunk 4 above)
Comment (mgorski98) [reply src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:25]: Fixed now (added an enum indicating type of figures to be drawn)

**Pull request closed without merging**